README.md

```diff
@@ -1,84 +1,103 @@
 
-## Install
+Configs matching a given schema are sourced from the following locations in the following order of precedence:
 
-You can grab source-configs from NPM with NPM or yarn:
+- Command line argument set in the schema via `commandLineArg`.
+- Environment variable set in the schema via `envVar`.
+- Deployment config file declared via:
+  - Command line argument: `--deploy-config` or `--dc`.
+  - Environment variable: `DEPLOY_CONFIG`.
+  - package.json as a `deployConfig` field.
+- Default defined in the schema.
 
-```
-npm install source-configs
-```
+## Usage
 
-or
+First declare `source-configs` as a dependency in your app.
 
-```
-yarn add source-configs
+Then create a schema object.
+
+A schema is an object that serves as a template for the desired final configuration consumed by your app.
+
+Here's a simple example declaring that our application will use a WebSocket with a configurable host, port, and protocol: 
+
+```js
+{
+  websocket: {
+    host: {},
+    port: {},
+    protocol: {}
+  }
+}
 ```
 
-## Priority basis
+### Schema metadata
 
-SourceConfigs will examine the 4 following locations and will fallback to the one below it if it doesn't find the config value
+Schemas support the following metadata for each configurable property in order to create constraints if desired or add additional functionality:
 
-* Command line arguments set in the schema under commandLineArg
-* Environment variable set in the schema as envVar
-* Deployment config file declared in the package.json as a `deployConfig` field, as a command line argument of `--deploy-file` or `--df`, or an environment variable of `SC_DEPLOY_FILE`.
-* Default defined in the schema
+- `description` *[String]*: Describe what this config will be used for.
+- `default` *[any]*: Set a default value for this config. If not set, the default is null.
+- `values` *[Array]*: Enumerated list of values that are valid. If not set, any value will be valid.
+- `commandLineArg` *[String]*: Command line argument to set this config. Will be converted to camelCase on the Node.js side. Example: the command line argument `--doThing` would be `doThing` on the Node.js side. If not set, source-configs will not listen for command line arguments to set the value for this config. [TODO: is the camelCaser really desirable?]
+- `envVar` *[String]*: Environment variable to set this config. If not set, source-configs will not listen for an environment variable to set the value for this config.
+- `envVarParser` *[Function]*: [TODO: simplify this] A function to parse environment variables or a string which will be used as a delimiter for simple delimiter split strings. This can only be implemented when writing the schema in a JS file or overwritten before being initialized.
 
-## Schema layout:
+[TODO: simplify this] Each configurable property can be a function which takes the parent scope config as a parameter to create strings based upon other config primitives.
```

Comment:
After thinking about this further, do we even need this? In the case that you are creating new config properties based on others, a developer should have access to the config as defined previously already so you could just create these variables in the buisness logic when needed.

Reply:
This was removed